### PR TITLE
feat: allow unique AggregationMode per Pragma pair

### PR DESCRIPTION
**Pull request by tserg**
This PR introduces a change to the Pragma adapter that lets the AggregationMode be set for each Pragma pair ID. This would allow the AggregationMode::ConversionRate option to be used for STRK LSTs, which is more robust and available.

Should we have a new `pragma_v2.cairo` file or is it safe to overwrite the existing `pragma.cairo` file?

### CODE REVIEW

File: src/types.cairo [hunk 1]
```diff
@@ -195,10 +195,12 @@ pub struct QuoteTokenInfo {
 //
 
 pub mod pragma {
-    #[derive(Copy, Drop, Serde)]
+    #[derive(Copy, Drop, PartialEq, Serde, starknet::Store)]
     pub enum AggregationMode {
+        #[default]
```
Comment (tserg) [inline src/types.cairo:200]: Defaults to Median mode. This is necessary for reading the settings of an address that has not been added.

**Comment by milancermak** [comment]
Definitely go with a new `pragma_v2.cairo` file. It's easier to understand for anyone coming into the project, building infra on top, etc.

File: src/types.cairo (hunk 1 above)
Comment (milancermak) [reply src/types.cairo:200]: > reading the settings of an address that has not been added

That would only happen by mistake right? E.g. calling `fetch_price` with a wrong yang address in which case the fn would fail with the assert we have in place.

**Review by milancermak — APPROVED** [review]

File: src/types.cairo (hunk 1 above)
Comment (tserg) [reply src/types.cairo:200]: It was actually caught by `test_update_prices_fails_with_no_yangs_in_seer` in the Seer test suite for calling `update_prices` without any yangs, so it is indeed quite an edge case. Should I remove the default?

Comment (milancermak) [reply src/types.cairo:200]: Nah, keep it, it's cool, I was just curious.

**Comment by tserg** [comment]
> Definitely go with a new pragma_v2.cairo file. It's easier to understand for anyone coming into the project, building infra on top, etc.

I moved v2 into a new `pragma_v2.cairo` file and restored the original version in `pragma.cairo` file. I have also moved the unit tests for v2 into a new standalone file so it exists in parallel with the unit tests for v1. For integration tests like in Seer, however, I switched from Pragma v1 to Pragma v2. I think this strikes a good balance between the need to retain Pragma v1 for tracing/background and maintainability of the test suite in relation to old code. What do you think?

**Comment by milancermak** [comment]
All good!
